File: app-java/src/main/java/net/mEmoZz/yts/java/ui/widgets/SwipeRefreshView.java
package net.mEmoZz.yts.java.ui.widgets;

import android.content.Context;
import android.support.v4.widget.SwipeRefreshLayout;
import android.util.AttributeSet;
import android.view.View;
import android.view.ViewGroup;
import net.mEmoZz.yts.java.R;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public class SwipeRefreshView extends SwipeRefreshLayout {

  private ViewGroup container;

  public SwipeRefreshView(Context context) {
    super(context);
  }

  public SwipeRefreshView(Context context, AttributeSet attrs) {
    super(context, attrs);
    setColorSchemeResources(R.color.colorAccent);
    setEnabled(false);
  }

  @Override public boolean canChildScrollUp() {
    ViewGroup container = getContainer();
    if (container == null) return false;

    View view = container.getChildAt(0);
    if (view.getVisibility() != View.VISIBLE) view = container.getChildAt(1);
    return view.canScrollVertically(-1);
  }

  private ViewGroup getContainer() {
    if (container != null) return container;
    for (int i = 0; i < getChildCount(); i++) {
      if (getChildAt(i) instanceof ViewGroup) {
        container = (ViewGroup) getChildAt(i);
        break;
      }
    }
    if (container == null) throw new RuntimeException("Container view not found");
    return container;
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/widgets/SwipeRefreshView.kt
package net.mEmoZz.yts.kotlin.ui.widgets

import android.content.Context
import android.support.v4.widget.SwipeRefreshLayout
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import net.mEmoZz.yts.kotlin.R

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class SwipeRefreshView @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null
) : SwipeRefreshLayout(context, attrs) {

  private var container: ViewGroup? = null

  init {
    setColorSchemeResources(R.color.colorAccent)
    isEnabled = false
  }

  override fun canChildScrollUp(): Boolean {
    val container = getContainer() ?: return false

    var view = container.getChildAt(0)
    if (view.visibility != View.VISIBLE) view = container.getChildAt(1)
    return view.canScrollVertically(-1)
  }

  private fun getContainer(): ViewGroup? {
    if (container != null) return container
    for (i in 0 until childCount) {
      if (getChildAt(i) is ViewGroup) {
        container = getChildAt(i) as ViewGroup
        break
      }
    }
    if (container == null) throw RuntimeException("Container view not found")
    return container
  }
}
